pyccel/ast/core.py

```diff
@@ -3764,6 +3711,93 @@ def __str__(self):
                 args   = ', '.join(self.args),
                 result = result)
 
+class Interface(Basic):
+
+    """Represent an Interface"""
+    def __new__( cls, *args, **kwargs ):
+        return Basic.__new__(cls)
+
+    def __init__(
+        self,
+        name,
+        functions,
+        hide=False,
+        is_argument = False,
+        ):
+
+        if not isinstance(name, str):
+            raise TypeError('Expecting an str')
+        if not isinstance(functions, list):
+            raise TypeError('Expecting a list')
+        self._name = name
+        self._functions = functions
+        self._hide = hide
+        self._is_argument = is_argument
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def functions(self):
+        return self._functions
+
+    @property
+    def hide(self):
```

Comment:
Can you add docstrings to some (ideally all) of these class methods please.
What is this function for?